### DISCUSSION TITLE
My ARM64 build generates an x86_64 build

### DISCUSSION THREAD

**Discussion by sontek** [pypa/cibuildwheel#1317] [Q&A]
Hello, I'm trying to get ARM64 builds published for M1 builds.  I have this github action:

https://github.com/sontek/tfparse/blob/fix-arm64-builds/.github/workflows/release.yml#L31-L59

When it runs it does:

```Building cp310-macosx_arm64 wheel
CPython 3.10 macOS arm64 - Apple Silicon

Installing Python cp310...

...

    Created wheel for tfparse: filename=tfparse-0.2.0-cp310-cp310-macosx_11_0_x86_64.whl size=6921462 sha256=6e2fa98c54725651d0d6bf1dc8f79e59530affc9c58f1d16a597278e91e5144a

```

When I expected an arm64 wheel.  Am I doing something wrong here?

This is the full action:

https://github.com/sontek/tfparse/actions/runs/3283882306/jobs/5409160527

**Comment by joerick** [comment]
I see that your build backend is poetry, and that is invoking setuptools in `build.py`. I haven't seen this setup before, but the first thing to check is the env vars we set for cross compilation - _PYTHON_HOST_PLATFORM and ARCHFLAGS - check they're reaching setuptools. I wonder if poetry is fiddling with them for some reason.

**Reply by sontek** [reply]
Thanks!   I'll dig into that to see if I can see any issues.   I also just attempted to rip out `setuptools` completely and I'm still seeing the problem:

https://github.com/sontek/tfparse/blob/fix-arm64-builds/build.py#L11-L42

With this build:
https://github.com/sontek/tfparse/actions/runs/3288591323/jobs/5419110550

Could it have something to do with `pyproject.toml` instead of `setup.py` driving things?     With `setuptools` completely ripped out I'd expect the wheel to do the right arch?

**Reply by sontek** [reply]
Looks like the problem might be related to:

https://github.com/python-poetry/poetry/issues/2051

?
